Chap_API_Reserved_Keys.tex

```diff
@@ -118,7 +118,8 @@ \subsection{Session realm attributes}
 }
 %
 \declareAttribute{PMIX_TDIR_RMCLEAN}{"pmix.tdir.rmclean"}{bool}{
-Resource Manager will cleanup assigned temporary directory trees.
+The Resource Manager will remove any directories or files it creates in
+\refattr{PMIX_TMPDIR}.
```

Comment:
Not quite correct, I believe. It means the RM will cleanup all the dirs and/or files that are placed in the PMIx session directories. In other words, the RM is going to cleanup the PMIx temp directory tree (i.e., the files _PMIx_ creates, not the RM), so PMIx itself doesn't need to worry about it.

Reply:
We were going for something more definitive than "cleanup", but admit we were not sure what exactly happens.   Would this be better (or can you suggest what would be most accurate?):

Reply:
So what happens is this: the host environment passes down the `PMIX_TMPDIR` which gives the PMIx server a root directory that is known "safe". The server then puts things like its shared memory segment and rendezvous files under that location. When the server finalizes, it normally will delete everything it put under `PMIX_TMPDIR`.

What this attribute declares is that the RM (i.e., the host) will delete this for us, usually by deleting the entire `PMIX_TMPDIR` directory tree. Thus, the PMIx server doesn't bother to delete anything - it just finalizes, leaving whatever files were created there.

Reply:
So should we say, "The Resource Manager will remove the directory \refattr{PMIX_TMPDIR} at server finalization time"?

Reply:
Well...not exactly. The RM will remove the directory whenever it decides to do it. It probably would do it once it finalizes the PMIx server library - but that's totally at their discretion. I would just leave it as "the RM will remove the directory".